### PR TITLE
Add precompiled gems for `mingw`

### PR DESCRIPTION
**Pull request by ianks**
This PR adds precompiled gems for `mingw`, so Windows users do not have to install a Rust toolchain to use this gem.

I took the opportunity to refactor CI a bit to make sure we only support stable Ruby versions (2.7..3.1 as of today).

### CODE REVIEW

**Comment by ianks** [comment]
Just waiting on https://github.com/matsadler/magnus/pull/27 now

File: .github/workflows/build-gems.yml [hunk 1]
```diff
@@ -7,18 +7,29 @@ on:
...
       matrix:
-        ruby-platform:
-          - "aarch64-linux"
-          - "arm64-darwin"
-          - "x86_64-darwin"
-          - "x86_64-linux"
-          - "x86_64-linux-musl"
+        ruby-platform: ${{ fromJSON(needs.ci-data.outputs.result).supported-ruby-platforms }}
+    env:
+      CROSS_PLATFORMS: ${{ matrix.ruby-platform }}
```
Comment (jbourassa) [inline .github/workflows/build-gems.yml:32]: What read this? Couldn't find any reference to it in the gems (`rg CROSS_PLATFORMS $(bundle show --paths)`) nor the actions.

File: .github/workflows/build-gems.yml [hunk 2]
```diff
@@ -7,18 +7,29 @@ on:
     branches: ["main", "cross-gem/*"]
 
 jobs:
+  ci-data:
+    runs-on: ubuntu-latest
+    outputs:
+      result: ${{ steps.fetch.outputs.result }}
+    steps:
+      - id: fetch
+        uses: oxidize-rb/actions/fetch-ci-data@main
```
Comment (jbourassa) [inline .github/workflows/build-gems.yml:16]: 👌 really cool

File: rakelib/compile.rake [hunk 3]
```diff
@@ -2,13 +2,7 @@ require "rake/extensiontask"
...
 GEMSPEC = Bundler.load_gemspec("wasmtime-rb.gemspec")
 
-CROSS_PLATFORMS = [
-  "aarch64-linux",
-  "arm64-darwin",
-  "x86_64-darwin",
-  "x86_64-linux",
-  "x86_64-linux-musl"
-]
+CROSS_PLATFORMS = [ENV["RUBY_TARGET"]].compact
```
Comment (jbourassa) [inline rakelib/compile.rake:5]: If I get this right, it should've been that way from the beginning, right? We run the  compile task once per target as per the matrix, and we get the [`RUBY_TARGET` from the Dockerfiles](https://github.com/oxidize-rb/rb-sys/blob/fa0f82885d9fff747af024288636cb6e4eaf4eed/docker/setup/rubybashrc.sh#L54).

**Review by jbourassa — APPROVED** [review]
Thanks for this. Looks like the build is failing, not sure why, it works on my M1 mac but fails on CI for arm64-darwin.

Left some questions so that I can learn 🤓.

File: .github/workflows/build-gems.yml (hunk 1 above)
Comment (ianks) [reply .github/workflows/build-gems.yml:32]: Ahh, it's an old diff that is not needed anymore. Will delete.

File: rakelib/compile.rake (hunk 3 above)
Comment (ianks) [reply rakelib/compile.rake:5]: Yup exactly. Seems like the sane default for GitHub actions, would be nice to have that be the default for RCD somehow...

**Pull request merged**